nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/PutSplunk.java

```diff
@@ -200,20 +174,15 @@ public void process(final InputStream in) throws IOException {
         activeBatches.add(messageBatch);
 
         // attempt to send the data and add the appropriate range
-        try {
-            sender.send(buf);
-            messageBatch.addSuccessfulRange(0L, flowFile.getSize());
-        } catch (IOException e) {
-            messageBatch.addFailedRange(0L, flowFile.getSize(), e);
-            context.yield();
-        }
+        sender.sendEvent(buf);
+        messageBatch.addSuccessfulRange(0L, flowFile.getSize());
     }
 
     /**
      * Read delimited messages from the FlowFile tracking which messages are sent successfully.
      */
     private void processDelimitedMessages(final ProcessContext context, final ProcessSession session, final FlowFile flowFile,
-                                          final ChannelSender sender, final String delimiter) {
+                                          final EventSender<byte[]> sender, final String delimiter) {
```

Comment:
Is it necessary to pass `EventSender` as parameter here since it is already a class member?